spyder/api/plugins/new_api.py

```diff
@@ -781,302 +1211,444 @@ def get_command_line_options(self):
+    """
+    Give the plugin focus on switch to plugin calls.
+
+    If ``True``, the plugin will be given focus when switched to.
+    If ``False``, the default, the plugin's widget will still be raised to
+    the foreground, but it will not be given focus until the switch action
+    is called a second time.
+    """
+
+    CAN_HANDLE_FILE_ACTIONS: bool = False
+    """
+    Whether the plugin declares it can handle file actions.
+
+    If set to ``True``, then the :meth:`create_new_file`,
+    :meth:`open_last_closed_file`, :meth:`save_file`, :meth:`save_file_as`,
+    :meth:`save_copy_as`, :meth:`save_all`, :meth:`revert_file`,
+    :meth:`close_file` and :meth:`close_all` methods will be called
+    to handle the corresponding actions.
+
+    If set to ``False``, the default, the corresponding actions fall back
+    to calling these methods on the :guilabel:`Editor`.
+
+    Individual actions can be disabled with the
+    :meth:`spyder.plugins.application.plugin.enable_file_action` method
+    in the :guilabel:`Application` plugin.
+    """
+
+    FILE_EXTENSIONS: list[str] = []
+    """
+    List of file extensions which the plugin can open.
+
+    If the user opens a file with one of these extensions, then the file
+    will open in this plugin using its :meth:`open_file` method.
+
+    Examples
+    --------
+
+    For example, in the Spyder-Notebook plugin to open Jupyter notebooks:
+
+    .. code-block:: python
+
+        FILE_EXTENSIONS = [".ipynb"]
+    """
+
+    CAN_HANDLE_EDIT_ACTIONS: bool = False
+    """
+    Whether the plugin can handle editing actions.
+
+    If set to ``True``, then the :meth:`undo`, :meth:`redo`, :meth:`cut`,
+    :meth:`copy`, :meth:`paste` and :meth:`select_all` methods will be called
+    to handle the corresponding actions.
+
+    If set to ``False``, the default, the corresponding actions fall back
+    to calling these methods on the :guilabel:`Editor`.
+
+    Individual actions can be disabled with the
+    :meth:`spyder.plugins.application.plugin.enable_edit_action` method
+    in the :guilabel:`Application` plugin.
+    """
+
     CAN_HANDLE_SEARCH_ACTIONS = False
+    """
+    Whether the plugin can handle search actions.
+
+    If set to ``True``, then the :meth:`find`, :meth:`find_next`,
+    :meth:`find_previous` and :meth:`replace` methods will be called
+    to handle the corresponding actions.
+
+    If set to ``False``, the default, the corresponding actions fall back
+    to calling these methods on the :guilabel:`Editor`.
+
+    Individual actions can be disabled with the
+    :meth:`spyder.plugins.application.plugin.enable_search_action` method
+    in the :guilabel:`Application` plugin.
+    """
 
     # ---- API: Available signals
     # -------------------------------------------------------------------------
     sig_focus_changed = Signal()
     """
-    This signal is emitted to inform the focus of this plugin has changed.
+    Report the focus state of this plugin has changed.
```

Comment:
Docstrings for all these signals have the same problems as the ones for `SpyderPluginV2`.

Reply:
Adding the type documentation here too